richtextfx/src/main/java/org/fxmisc/richtext/model/SuperCodec.java

```diff
@@ -0,0 +1,51 @@
+package org.fxmisc.richtext.model;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public interface SuperCodec<S, T extends S> extends Codec<T> {
```

Comment:
Its used in `StyleClassedTextArea`: 

    SuperCodec.upCast(SuperCodec.collectionListCodec(Codec.STRING_CODEC)),
    SuperCodec.upCast(SuperCodec.collectionListCodec(Codec.STRING_CODEC))